bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/transactions/TransactionResourceManager.java

```diff
@@ -96,6 +118,51 @@ public static TransactionResourceManager getInstance() {
         return transactionResourceManager;
     }
 
+    /**
+     * This method sets values for atomikos transaction log path and name properties using the available configs.
+     *
+     */
+    private void setLogProperties() {
+        final Path projectRoot = findProjectRoot(Paths.get(System.getProperty("user.dir")));
+        if (projectRoot != null) {
+            String logDir = getTransactionLogDirectory();
+            String logPath = projectRoot.toAbsolutePath().toString() + "/" + logDir;
+            Path transactionLogDirectory = Paths.get(logPath);
+            if (!Files.exists(transactionLogDirectory)) {
+                try {
+                    Files.createDirectory(transactionLogDirectory);
+                } catch (IOException e) {
+                    stderr.println("error: failed to create transaction log directory");
```

Comment:
Improve the error msg to include the directory

Reply:
Fixed in 91ca624